client/my-sites/stats/controller.jsx

```diff
@@ -209,11 +198,12 @@ module.exports = {
 		}
 		siteId = currentSite ? ( currentSite.ID || 0 ) : 0;
 
-		const activeFilter = find( filters(), ( filter ) => {
+		const filters = getSiteFilters( siteId );
+		const activeFilter = find( filters, ( filter ) => {
 			return context.pathname === filter.path || ( filter.altPaths && -1 !== filter.altPaths.indexOf( context.pathname ) );
 		} );
 
-		if ( ! siteFragment || ! activeFilter ) {
+		if ( ! activeFilter ) {
 			next();
 		} else {
 			if ( 0 === siteId ) {
```

Comment:
Actually, I was thinking of leaving all the sites list cleaning from the controller in a separate PR after this PR and #11162 land